onedal/linear_model/linear_model.py

```diff
@@ -193,12 +166,66 @@ def __init__(
         super().__init__(fit_intercept=fit_intercept, copy_X=copy_X, algorithm=algorithm)
 
     def fit(self, X, y, queue=None):
-        return super()._fit(
-            X, y, self._get_backend("linear_model", "regression", None), queue
-        )
+        """
+        Fit linear model.
+        Parameters
+        ----------
+        X : {array-like, sparse matrix} of shape (n_samples, n_features)
+            Training data.
 
-    def predict(self, X, queue=None):
-        y = super()._predict(
-            X, self._get_backend("linear_model", "regression", None), queue
+        y : array-like of shape (n_samples,) or (n_samples, n_targets)
+            Target values. Will be cast to X's dtype if necessary.
+
+        queue : dpctl.SyclQueue
+            If not None, use this queue for computations.
+
+        Returns
+        -------
+        self : object
+            Fitted Estimator.
+        """
+        module = self._get_backend("linear_model", "regression")
+
+        if X.shape[1] + int(self.fit_intercept) > X.shape[0]:
+            raise TypeError("Not enough samples to fit")
```

Comment:
Leads to fail of previously successful linear regression spmd example. This condition should be only raised in cases where it will certainly fail (ie only incremental?)

Reply:
Our C++ implementation is unstable if number of samples does not satisfy this condition and it was even stated before in the code, but only in sklearnex part
https://github.com/intel/scikit-learn-intelex/blob/263b9c17475314689498539b45b3d63c16454e09/sklearnex/linear_model/linear.py#L244
so I decided that it is better to add this check here as well to make this limitation more clear until it will be well investigated. JIRA task about investigation has been created